### PR TITLE
Gift card emails

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds mailing about new gift card to customer.

TO DO:
* [x] User email plugin update
* [x] Update sendgrid
* [x] New event when card has been sent
* [x] Add template to sample repo - https://github.com/saleor/saleor-email-templates/pull/4
* [x] Update docs - https://github.com/mirumee/saleor-docs/pull/299

⚠️ This is a PR to feature branch.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3901 Create task for sending gift card to the customer](https://app.clickup.com/t/2549495/SALEOR-3901)

File: saleor/plugins/sendgrid/tests/test_tasks.py [hunk 1]
```diff
@@ -52,7 +55,7 @@ def test_send_email(
     mocked_api_client, mocked_mail, sendgrid_email_plugin, sample_payload
 ):
     plugin = sendgrid_email_plugin(
-        sender_name="Sander Name",
+        sender_name="Sender Name",
```
Comment (korycins) [inline saleor/plugins/sendgrid/tests/test_tasks.py:58]: Nice catch 😄

**Review by korycins — COMMENTED** [review]
Looks good. Remember about updating saleor-templates repository and docs.

File: saleor/graphql/giftcard/mutations.py [hunk 2]
```diff
@@ -99,7 +100,7 @@ def clean_input(cls, info, instance, data):
         cls.clean_expiry_settings(cleaned_input, instance)
         cls.clean_balance(cleaned_input, instance)
 
-        cleaned_input.get("user_email", None)
+        # TODO: validate user email
```
Comment (fowczarek) [inline saleor/graphql/giftcard/mutations.py:103]: It will be done in separate PR?

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/giftcard/mutations.py (hunk 2 above)
Comment (tomaszszymanski129) [reply saleor/graphql/giftcard/mutations.py:103]: Yep, email should be validated in couple of these mutations, will be added later.

**Review by maarcingebala — APPROVED** [review]

File: saleor/giftcard/notifications.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        "requester": {
+            "user_id": user.id if user else None,
+            "email": user.email if user else None,
+            "app_id": app.id if app else None,
+            "app_name": app.name if app else None,
+        },
+        "recipient_email": email,
+        "gift_card": {"id": gift_card.id, "code": gift_card.code},
+        **get_site_context(),
+    }
```
Comment (IKarbowiak) [inline saleor/giftcard/notifications.py:17]: Also `balance` should be attached. I also wondering about changing `email` to `name` and return the user `name` if the user instance exists, user email otherwise.
